extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -6,7 +6,44 @@ public function appreciate() {
 		global $wgUser;
 
 		$this->request->assertValidWriteRequest( $wgUser );
-		//TODO: do something with apprecation
+
+		$revisionId = $this->request->getInt( 'revision' );
+		$revision = Revision::newFromId( $revisionId );
+		if ( $revision instanceof Revision ) {
+			( new RevisionUpvotesService() )->addUpvote(
+				$this->wg->CityId,
```

Comment:
You use globals inconsistently in this method (`CityId` and `User`), please choose one approach and stick to it